src/ceph-disk/ceph_disk/main.py

```diff
@@ -919,7 +919,7 @@ def is_mounted(dev):
             path = fields[1]
             if os.path.isabs(mounts_dev) and os.path.exists(mounts_dev):
                 mounts_dev = os.path.realpath(mounts_dev)
-                if mounts_dev == dev:
+                if _bytes2str(mounts_dev) == dev:
```

Comment:
` os.path.realpath()` returns `str` on both python3 and python2. so `dev` here is a `str`, the same applies to `mounts_dev`. @tserong am i missing anything?

Reply:
Seems it returns whatever type was passed in:

````
python3
Python 3.6.5 (default, Mar 31 2018, 19:45:04) [GCC] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> s = '/dev/sda1'
>>> type(os.path.realpath(s))
<class 'str'>
>>> b = bytes('/dev/sda1', 'utf8')
>>> type(os.path.realpath(b))
<class 'bytes'>
````

Reply:
ic. python3 reads a file opened with `rb` as `bytes`, if the file is opened with `r`, the returned file object will give us `str` lines.

but the open `mode` does not have same effect on python2.

@tserong based on your finding and my analysis above, i think a safer change would be

1. normalize the `dev` parameter to a `str` using `_bytes2str` before passing it to `os.path.realpath()`, so we don't need to audit all the callers of this function to make sure that they are passing a `str` 
1. use `r` not `rb` as the `open` mode. so we can be immune to this subtle difference between python2 and python3.

what do you think?

Reply:
That sounds like a good approach, will do.